
<launch>
  <env name="ROSCONSOLE_FORMAT" value="[${severity}][${thread}][${node}/${function}:${line}]: ${message}"/>
  <env name="ROS_LOG_DIR" value="/tmp/"/>

  <arg name="header_frame_id" default="wheel_controller" />

  <arg name="map_frame" default="map" />
  <arg name="base_frame" default="base_stabilized"/>
  <arg name="odom_frame" default="base_frame"/>
  <arg name="pub_map_odom_transform" default="true"/>
  <arg name="map_size" default="2048"/>
  <arg name="scan_subscriber_queue_size" default="5"/>
  <param name="/use_sim_time" value="false"/>

  <arg name="scan_topic" default="scan"/>
  <arg name="tf_map_scanmatch_transform_frame_name" default="odometry"/>
  <arg name="geotiff_map_file_path" default="/tmp/maps"/>

  <arg name="battery_state_output_topic" default="/robot_platform/battery" />
  <arg name="gps_state_output_topic" default="/robot_platform/gps"/>
  <arg name="imu_state_output_topic" default="/robot_platform/imu"/>
  <arg name="odom_state_output_topic" default="/robot_platform/odom"/>
  <arg name="wheel_positions_input_topic" default="/robot_platfor+m/wheel_request"/>
  <arg name="platform_status_output_topic" default="/robot_platform/status"/>

  <node pkg="robot_platform" type="wheel_controller" name="wheel_controller" output="screen" respawn="true">

    <param name="serial_dev" type="string" value="/dev/serial/by-id/usb-Teensyduino_USB_Serial_13671670-if00" />
    <param name="serial_baudrate" type="int" value="115200" />

    <param name="header_frame_id" type="string" value="$(arg header_frame_id)" />
    
    <param name="raw_input_topic" type="string" value="/robot_platform/raw_serial_request" />
    <param name="raw_output_topic" type="string" value="/robot_platform/raw_serial_response" />
    
    <param name="battery_state_output_topic" type="string" value="$(arg battery_state_output_topic)" />
    <param name="gps_state_output_topic" type="string" value="$(arg gps_state_output_topic)" />
    <param name="imu_state_output_topic" type="string" value="$(arg imu_state_output_topic)" />
    <param name="wheel_positions_input_topic" type="string" value="$(arg wheel_positions_input_topic)" />
    <param name="platform_status_output_topic" type="string" value="$(arg platform_status_output_topic)" />
    <param name="odometry_output_topic" type="string" value="$(arg odom_state_output_topic)" />
  </node>

</launch>
